index.js

```diff
@@ -1,18 +1,54 @@
 import React from 'react';
-import {AppRegistry, StyleSheet, Text, View} from 'react-native';
+import {AppRegistry, StyleSheet, Text, View, Button, NativeModules} from 'react-native';
 
 class RNHighScores extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      scores: props.scores
+    };
+
+    this._exitButtonPressed = this._exitButtonPressed.bind(this);
+    this._clearButtonPressed = this._clearButtonPressed.bind(this);
+  }
+
+  _exitButtonPressed() {
+    NativeModules.F3HBridgeModule.exitHighScoreButtonTapped(this.state.scores)
+  }
+
+  _clearButtonPressed() {
+    this.setState({
+      scores: []
+    })
+  }
+
   render() {
-    var contents = this.props['scores'].map((score) => (
-      <Text key={score.name}>
-        {score.name}:{score.value}
-        {'\n'}
+    var scores = this.state.scores;
+
+    var scoreContent = scores.map((score, i) => (
+      <Text key={'score-' + i}>
+        {score.value} - {score.createdAt}
+        {"\n"}
       </Text>
-    ));
+    ))
+
+    var contents = (scores.length > 0) ?
+      scoreContent
+    :
+      <Text>No scores yet!</Text>
+    ;
+
     return (
       <View style={styles.container}>
         <Text style={styles.highScoresTitle}>2048 High Scores!</Text>
         <Text style={styles.scores}>{contents}</Text>
+        <Button title={'Clear'} onPress={this._clearButtonPressed}>
+          <Text>Clear</Text>
+        </Button>
+        <Button title={'Exit'} onPress={this._exitButtonPressed}>
+          <Text>Exit</Text>
+        </Button>
```

Comment:
Similar to above, if I was doing a native iOS UI I'd probably put these buttons on a navigation bar, with clear on the top left and Done on the top right. I have no idea how to set this up in React Native. But that's what we're here to learn.

Reply:
There's more than one way to do this. I'm not saying we shouldn't do it, just that it might not be done before Tuesday.

https://wix.github.io/react-native-navigation

Reply:
To be clear, I'm talking about using the UINavigationController to get a "buttons and title" bar at the top of the screen; on iOS it is common to use it even if you aren't actually doing any navigating.

So I'm not asking how we do navigation at this stage (I think that's out of scope), I'm asking how we do the cross-platform React Native equivalent. Hopefully it will be a case of asking for a <TopBar> or something like that.

Reply:
No I understand. But navigation isn't baked into the framework, so this is something that needs to be installed (even just for the navbar with no navigation) and there is more than one option.

I looked into installing one this morning and it's giving me an error. I'm going to say we're out of time to add this during this cycle.

Reply:
We shouldn't need a navigation framework to get a UINavigationBar, but now I am realizing just how little help React Native is going to be as far as sticking to platform conventions. I assumed it was going to be like the (ancient) Java Window Toolkit and it uses the appropriate parts (menu, buttons, etc.) on the platform you're running on. But React Native is more like Java's Swing, where it just draws its own UI. It's using native components, but it isn't assembling them the way you would if you were writing a 100% native app.